client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -496,7 +496,7 @@ public void flush() throws SegmentSealedException {
         if (numInflight != 0) {
             try {
                 ClientConnection connection = Futures.getThrowingException(getConnection());
-                connection.send(new KeepAlive());
+                connection.send(new Flush(writerId, segmentName));
```

Comment:
If I understand it right, this flush command is invoking `breakFromAppend` in `CommandEncoder` rather than send a command over the wire. Is there a way of invoking `breakFromAppend` under the same condition without having to create a new command?

Getting rid of  `connection.send(new KeepAlive())` in line 499 sounds right.

Reply:
No, we dont have any such command ; there is Event Command , but we need a command which encapsulates the segment and writer , so that we dont flush other writers data.